src/upnp.cpp

```diff
@@ -698,8 +698,8 @@ void upnp::post(upnp::rootdevice const& d, char const* soap
 		"Soapaction: \"%s#%s\"\r\n\r\n"
 		"%s"
 		, d.path.c_str(), d.hostname.c_str(), d.port
-		, int(strlen(soap)), d.service_namespace.c_str(), soap_action
-		, soap);
+		, int(soap.length()), d.service_namespace.c_str(), soap_action
+		, soap.c_str());
```

Comment:
the way to print a ``string_view`` without allocating a new string is like this: ("%*s", int(soap.size()), soap.data())